ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -238,24 +239,40 @@ def _not_done_training(self) -> bool:
             or not self.train_model
         )
 
-    def write_to_tensorboard(self, global_step: int) -> None:
+    def write_to_tensorboard(self, global_step: int) -> Dict[str, Dict[str, float]]:
+        stats_out = {}
         for brain_name, trainer in self.trainers.items():
             # Write training statistics to Tensorboard.
             delta_train_start = time() - self.training_start_time
             if self.meta_curriculum is not None:
-                trainer.write_summary(
+                brain_stats = trainer.write_summary(
                     global_step,
                     delta_train_start,
                     lesson_num=self.meta_curriculum.brains_to_curriculums[
                         brain_name
                     ].lesson_num,
                 )
             else:
-                trainer.write_summary(global_step, delta_train_start)
+                brain_stats = trainer.write_summary(global_step, delta_train_start)
+            stats_out[brain_name] = brain_stats
```

Comment:
There's currently 2 (now 3) ways we get the reward out of the trainer - `_get_measure_vals()` for meta curriculum, the `trainer.write_summary`, and however `trainer_metrics` does it. I wonder if we could make `_get_measure_vals()` more generic and use that to get the brain_stats instead of relying on write_summary (or the other way around)?

Reply:
OK, undid these changes and updated `_get_measure_vals()` to get the average of the rewards buffers, and call that on the test. I'm still not sure this is a great solution, since that buffer will only be length 1 for non-curriculum learning. What do you think?